tests/ruby/stats_spec.rb

```diff
@@ -25,9 +25,42 @@
         sleep(15.5)
         admin_conn = PG::connect(processes.pgcat.admin_connection_string)
         results = admin_conn.async_exec("SHOW STATS")[0]
-        admin_conn.close
+
+        # BEGIN, SELECT, COMMIT for each connection
+        expect(results["total_query_count"].to_i).to eq(9)
         expect(results["total_query_time"].to_i).to be_within(200).of(750)
-        expect(results["avg_query_time"].to_i).to be_within(50).of(250)
+        expect(results["avg_query_time"].to_i).to be_within(15).of(250.0 / 3)
+
+        # Single transaction for each connection
+        expect(results["total_xact_count"].to_i).to eq(3)
+        expect(results["total_xact_time"].to_i).to be_within(200).of(750)
+        expect(results["avg_xact_time"].to_i).to be_within(50).of(250)
+
+        expect(results["total_wait_time"].to_i).to_not eq(0)
+        expect(results["avg_wait_time"].to_i).to_not eq(0)
+
+        connections = Array.new(3) { PG::connect("#{pgcat_conn_str}?application_name=one_query") }
+        connections.each do |c|
+          Thread.new { c.async_exec("SELECT pg_sleep(0.125);") }
+        end
+        sleep(1)
+        connections.map(&:close)
+        # wait for averages to be calculated again
+        sleep(15.5)
+
+        results = admin_conn.async_exec("SHOW STATS")[0]
+        admin_conn.close
+        # Three more queries
+        expect(results["total_query_count"].to_i).to eq(12)
+        expect(results["total_query_time"].to_i).to be_within(225).of(1125)
+        # Average is 125ms in the current stats period
+        expect(results["avg_query_time"].to_i).to be_within(25).of(125.0)
+
+        # Autocommit transactions still increase the count and time
```

Comment:
This was already the case, and is the same as PgBouncer: https://github.com/pgbouncer/pgbouncer/blob/e2a2a682fe9069366fca970ec2492d774b4f6b40/test/test_admin.py#L74-L75